java/mykola-melnyk/src/com/zoolatech/lecture6/tasks/_2/Task2.java

```diff
@@ -0,0 +1,38 @@
+package com.zoolatech.lecture6.tasks._2;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import static com.zoolatech.lecture6.tasks._2.Type.STORE;
+import static com.zoolatech.lecture6.tasks._2.Type.WEBSITE;
+import static java.util.stream.Collectors.averagingDouble;
+
+/**
+ * Create a method that accepts a list of orders and returns a map of an average price of all store orders
+ * per each country. Use Stream API to do this. The order has such properties: string id,
+ * enum type (WEBSITE, STORE), float price (yeah, I know, but let’s simplify the code :)),
+ * string country. How would you change your code if a list contains duplicate items with the same id?
+ * Input: order1(“1”, “store”, 1.23, “Ukraine”), order2(“2”, “store”, 4.56, “Ukraine”),
+ * order3(“3”, “store”, 1.23, “USA”), order4(“4”, “website”, 1.23, “USA”)
+ * Output: “USA” -> 1.23, “Ukraine” -> 2.89
+ */
+
+public class Task2 {
+    public static Map<String, Double> grouping(Stream<Order> stream) {
```

Comment:
prefer to pass a collection into a method instead of a stream, and create a stream inside of the method (the stream can be used only once, and somebody might pass a closed stream into a method)

Reply:
Done.